proxy/http/HttpTransact.cc

```diff
@@ -7359,8 +7359,9 @@ HttpTransact::handle_server_died(State *s)
     body_type = "response#bad_response";
     break;
   case CONNECTION_ERROR:
-    status    = HTTP_STATUS_BAD_GATEWAY;
-    reason    = (char *)get_error_string(s->cause_of_death_errno);
+    status = HTTP_STATUS_BAD_GATEWAY;
+    reason =
+      static_cast<const char *>(get_error_string(s->cause_of_death_errno == 0 ? -ENET_CONNECT_FAILED : s->cause_of_death_errno));
```

Comment:
Why did we need the cast here?

Reply:
I may have blindly kept the existing cast, but changed it to a static_cast.
It does not appear that it was needed in the first place.
Happy to remove it in another PR if desired.